playroom/components/animation.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import Lottie from 'lottie-react';
+
+import type {LottieComponentProps} from 'lottie-react';
+import type {ExclusifyUnion} from '../../src/utils/utility-types';
+
+type Props = Omit<LottieComponentProps, 'animationData'> &
+    ExclusifyUnion<{animationData: unknown} | {animationUrl: string}>;
+
+const Animation = ({animationData: animationDataProp, animationUrl, ...props}: Props): JSX.Element | null => {
+    const [animationData, setAnimationData] = React.useState<unknown>();
+
+    React.useLayoutEffect(() => {
+        if (animationUrl) {
+            fetch(animationUrl)
+                .then((response) => {
+                    if (response.ok) {
+                        response.json().then((jsonData) => setAnimationData(jsonData));
+                    }
+                })
+                .catch(() => {
+                    setAnimationData(undefined);
+                });
+        } else {
+            setAnimationData(animationDataProp);
+        }
+    }, [animationUrl, animationDataProp]);
+
+    return animationData ? <Lottie animationData={animationData} {...props} /> : null;
+};
+
+export default Animation;
```

Comment:
why not something like this? we already have a Loader component in Playroom:
```suggestion
const Animation = ({animationData, animationUrl, ...props}: Props): JSX.Element | null => {
    if (animationData) {
        return <Lottie animationData={animationData} {...props} />;
    }
    return (
        <Loader
            load={animationUrl}
            render={(animationData) => <Lottie animationData={animationData} {...props} />}
        />
    );
};

export default Animation;
```

Reply:
I forgot completely about the Loader component, this is a perfect use case for it. Good catch!

Done :)